app/services/gmail/unread.py

```
@@ -0,0 +1,326 @@
+            # Progress: 40-100% for processing
+            progress = 40 + int((affected / total_emails) * 60)
+            state.update_unread_action_status(
+                affected_count=affected,
+                progress=progress,
+                message=f"Processed {affected}/{total_emails} emails...",
+            )
+            # Rate limiting for large batches
+            if i + batch_size < total_emails:
+                time.sleep(0.2)
+    except Exception as e:
+        logger.exception(f"Error during {action_name}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)